### ISSUE TITLE
Memory leak for mysql

### ISSUE THREAD

**Issue by thenik** [CollaboratingPlatypus/PetaPoco#721]
Hi,

Memory leak detected for mysql data via PetaPoco during intensive usage :
![image](https://github.com/user-attachments/assets/ebd1121c-7439-4637-84df-34e180657598)

Any ideas how to adjust it?

**Comment by asherber** [comment]
@thenik It's not clear to me what that image is showing. Can you please describe what you think is being leaked, and provide a minimal repro?

**Comment by thenik** [comment]
This images shows 1 of many reteined objects for my app during 1h of usage. 
For this image it is RuntimeMethodHandle .
About 26K  RuntimeMethodHandle objects retained for my app during 1h .
![image](https://github.com/user-attachments/assets/dfaa92aa-ab45-4390-8488-7881732af9cd)

I used dotMemory for analysis and as you see this RuntimeMethodHandle object came from PocoData.
I do not have an independent code sample that will reproduce that now.

**Comment by asherber** [comment]
If I had to guess, I'd say that this is something being kept in PocoData's cache rather than something being leaked. Are you running queries with thousands of different types?

Without some sort of minimal repro, this will be hard to diagnose.

**Comment by thenik** [comment]
@asherber Yes, you are right that I found something interesting with PocoData's cache. I call it memory leak =)

>>> Are you running queries with thousands of different types?
no =) I use for this case just 6 types, I marked them on the next image:

![image](https://github.com/user-attachments/assets/4ac8498f-45e6-4ad2-b35d-bfc2498e6e74)

to be honest I tried to create a demo c# app that will reproduce that, but still without success
1st version was 1 thread version with 1 type
2nd - multithread with 1 type
3d - multithread with 2 types

and there is no my "memory leak"
I do not have idea how to implement an app that will reproduce that now.

if any ideas what I can implement for demo app to reproduce or where to look - welcome.
Thanks
